Ginger/GingerCoreNET/SourceControl/SourceControlIntegration.cs

```diff
@@ -67,12 +68,14 @@ public static bool DeleteFile(SourceControlBase SourceControl, string path)
         public static bool UpdateFile(SourceControlBase SourceControl, string path)
         {
             string error = string.Empty;
-
+            RepositoryFolderBase repositoryFolderBase = WorkSpace.Instance.SolutionRepository.GetRepositoryFolderByPath(Path.GetDirectoryName(path));
+            repositoryFolderBase.PauseFileWatcher();
             if (!SourceControl.UpdateFile(path, ref error))
             {
                 Reporter.ToUser(eUserMsgKey.GeneralErrorOccured, error);
                 return false;
```

Comment:
if update file will fail code will return false. And file watcher will remain paused forever